File: selenium-webdriver-python-unittest/pages/payment_page.py
from pages.base_page import BasePage

from selenium import webdriver
from selenium.webdriver.support.ui import Select

class PaymentPage(BasePage):

  def enter_holder_name(self, holder_name):
    self.driver.find_element_by_name("holder_name").send_keys(holder_name)

  def enter_card_number(self, card_number):
    self.driver.find_element_by_name("card_number").send_keys(card_number)

  def enter_expiry_month(self, expiry_month):
    Select(self.driver.find_element_by_name("expiry_month")).select_by_visible_text(expiry_month)

  def enter_expiry_year(self, expiry_year):
    Select(self.driver.find_element_by_name("expiry_year")).select_by_visible_text(expiry_year)

  def click_pay_now(self):
    self.driver.find_element_by_xpath("//input[@value='Pay now']").click()
 
  def select_card_type(self, card_type):
    self.driver.find_element_by_xpath("//input[@name='card_type' and @value='" + card_type + "']").click()
  

File: selenium-webdriver-rspec/pages/payment_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class PaymentPage < AbstractPage

  def initialize(driver)
    super(driver, "") # <= TEXT UNIQUE TO THIS PAGE
  end

  def enter_holder_name(holder_name)
    driver.find_element(:name, "holder_name").send_keys(holder_name)
  end

  def enter_card_number(card_number)
    driver.find_element(:name, "card_number").send_keys(card_number)
  end

  def enter_expiry_month(expiry_month)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "expiry_month")).select_by(:text, expiry_month)
  end

  def enter_expiry_year(expiry_year)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "expiry_year")).select_by(:text, expiry_year)
  end

  def click_pay_now
    driver.find_element(:xpath,"//input[@value='Pay now']").click
  end

  def select_card_type(card_type)
    driver.find_element(:xpath, "//input[@name='card_type' and @value='" + card_type +  "']").click
  end

end
